tests/test_requests.py

```diff
@@ -1452,3 +1453,17 @@ def test_vendor_aliases():
 
     with pytest.raises(ImportError):
         from requests.packages import webbrowser
+
+def test_chunked_upload():
+    """can safely send generators"""
+    block_server = threading.Event()
```

Comment:
This is a really tiny note, but I think that it would be better if we called this `close_server`. It just feels a bit clearer to me.